### PR TITLE
delete useless ObjectMeta field from subset

### PR DESCRIPTION
**Pull request by kadisi**
Signed-off-by: kadisi <iamkadisi@163.com>

### CODE REVIEW

File: pkg/controller/uniteddeployment/subset.go [hunk 1]
```diff
@@ -24,15 +24,15 @@ import (
 
 // Subset stores the details of a subset resource owned by one UnitedDeployment.
 type Subset struct {
```
Comment (hantmac) [inline pkg/controller/uniteddeployment/subset.go:26]: In my opinion, using `metav1.ObjectMeta` is better than using specific param of ObjectMeta.

Comment (kadisi) [reply pkg/controller/uniteddeployment/subset.go:26]: `metav1.ObjectMeta` is only  used for `func (m *SubsetControl) objectKey() function `,  and too many fields need to be assigned to it in order to use it，
I think this one adds complexity and Subset is also not a real K8s object

Comment (hantmac) [reply pkg/controller/uniteddeployment/subset.go:26]: It may be that the developer considers scalability more important

Comment (Fei-Guo) [reply pkg/controller/uniteddeployment/subset.go:26]: A compromise is that we can leave most of the fields in subset.ObjectMeta empty/unassigned such as label, annotation, finalizier etc to reduce the cost of initialization. We can add initialization code for those fields later when we need them in new features. Of course, add a comment to explain the intention.   

@wu8685 @kadisi what do you think?

Comment (wu8685) [reply pkg/controller/uniteddeployment/subset.go:26]: Another choice is to use [GetObjectMeta()](https://github.com/openkruise/kruise/blob/master/pkg/controller/uniteddeployment/adapter/advanced_statefulset_adapter.go#L51) in each adapter, if we just want to get rid of the initialization.

**Comment by Fei-Guo** [comment]
It seems that the cleanup does not get a consensus so let us hold it. Please reopen if people changed their minds.

**Pull request closed without merging**